src/borg/testsuite/locking.py

```diff
@@ -10,6 +12,8 @@
 
 ID1 = "foo", 1, 1
 ID2 = "bar", 2, 2
+RACE_TEST_NUM_THREADS = 40
+RACE_TEST_DURATION = 0.4  # seconds
```

Comment:
was this enough to see test fails with the original code?

Reply:
Yes. On my laptop, at least, which was a fast machine...6 years ago. :-) We can increase that, if you deem it necessary, currently, I have no access to a fast machine. The probability should be very close to 1 to reveal the problem, so maybe there should be about 10 cases with more than two threads "thinking" they own the lock.